src/dmd/constfold.d

```diff
@@ -1358,13 +1378,10 @@ extern (C++) UnionExp Slice(Type type, Expression e1, Expression lwr, Expression
     else if (e1.op == TOK.arrayLiteral && lwr.op == TOK.int64 && upr.op == TOK.int64 && !hasSideEffect(e1))
     {
         ArrayLiteralExp es1 = cast(ArrayLiteralExp)e1;
-        uinteger_t ilwr = lwr.toInteger();
-        uinteger_t iupr = upr.toInteger();
-        if (iupr > es1.elements.dim || ilwr > iupr)
-        {
-            e1.error("array slice `[%llu .. %llu]` is out of bounds", ilwr, iupr);
```

Comment:
What happens with this error?

Reply:
> The error message printing turns out to be dead code

... from the description in this PR, I believe.